src/ng/compile.js

```diff
@@ -2644,16 +2639,12 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             break;
         }
       });
-      var destroyBindings = onNewScopeDestroyed ? function destroyBindings() {
+
+      return onNewScopeDestroyed && function destroyBindings() {
         for (var i = 0, ii = onNewScopeDestroyed.length; i < ii; ++i) {
           onNewScopeDestroyed[i]();
         }
-      } : noop;
-      if (newScope && destroyBindings !== noop) {
-        newScope.$on('$destroy', destroyBindings);
```

Comment:
I moved this out to the 1 of 3 `initializeDirectiveBindings` invocations that actually need it. I figured now that `initializeDirectiveBindings` binds to any `destination` and not specifically a scope this logic no longer belonged here...